src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -208,6 +209,56 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+         /// <summary>
+        /// Handles a request to get an execution plan
+        /// </summary>
+        public async Task HandleExecutionPlanRequest(QueryExecutionPlanParams planParams,
+            RequestContext<QueryExecutionPlanResult> requestContext)
+        {
+            try
+            {
+                // Attempt to load the query
+                Query query;
+                if (!ActiveQueries.TryGetValue(planParams.OwnerUri, out query))
+                {
+                    await requestContext.SendResult(new QueryExecutionPlanResult
+                    {
+                        Message = SR.QueryServiceRequestsNoQuery
+                    });
+                    return;
+                }
+
+                // Retrieve the requested execution plan and return it
+                var result = new QueryExecutionPlanResult
+                {
+                    Message = null,
+                    ExecutionPlan = await query.GetExecutionPlan(planParams.BatchIndex, planParams.ResultSetIndex)
+                };
+                await requestContext.SendResult(result);
+            }
+            catch (InvalidOperationException ioe)
+            {
+                // Return the error as a result
+                await requestContext.SendResult(new QueryExecutionPlanResult
+                {
+                    Message = ioe.Message
+                });
+            }
+            catch (ArgumentOutOfRangeException aoore)
+            {
+                // Return the error as a result
+                await requestContext.SendResult(new QueryExecutionPlanResult
+                {
+                    Message = aoore.Message
+                });
+            }
+            catch (Exception e)
```

Comment:
In general, I think we can have the other exception types just be caught using the general Exception catch and return all of them via SendError